### PR TITLE
[Fix] Apply partial MRoPE to Qwen3.5 attention

### PR DESCRIPTION
**Pull request by AntonyMei**
# Description

Before this fix, partial_rotary_factor is not applied in Qwen3OmniMoeThinkerTextRotaryEmbedding, resulting in repetition / corrupted output during generation. 

Added a long prompt test to `forward_pass_logit_checker.py` to catch such RoPE related issues in future

Added support for reading pkl files in `forward_pass_logit_checker.py`

FIXES: b/543115324

# Tests

- Ran this fix with vllm serve on a local TPU VM v5p-8 and ensures logits matched.
- Added a long prompt test to `forward_pass_logit_checker.py` to catch such RoPE related issues in future

```
P="The city of Lyon sits at the confluence of the Rhone and Saone rivers in eastern France. Founded as a Roman colony in 43 BC, it served for centuries as the capital of the Gauls. Its silk industry flourished in the sixteenth century, drawing weavers from across Europe. The old town preserves narrow covered passageways known as traboules, built so that silk merchants could carry bolts of fabric between workshops without exposing them to rain. "
PROMPT="$P$P$P$P$P"$'\n\n'"Based on the passage above, the traboules of Lyon were originally built so that"

python3 -m tests.assets.logits_generation.generate_hf_golden_logits \
  --model-id=Qwen/Qwen3.5-35B-A3B \
  --hf-model-path=<> \
  --hf-load-dtype=float32 \
  --prompts="$PROMPT" \
  --output-path=/mnt/disks/persist/golden_qwen35_longprompt.pkl \
  --output-format=pickle

python3 -m tests.utils.forward_pass_logit_checker src/maxtext/configs/base.yml \
  model_name=qwen3.5-35b-a3b tokenizer_path=<> \
  load_parameters_path=gs://maxtext-model-checkpoints/qwen3.5-35b-a3b/unscanned/0/items \
  scan_layers=false enable_nnx=true pure_nnx_decoder=true \
  use_multimodal=false per_device_batch_size=1 dtype=bfloat16 weight_dtype=bfloat16 \
  max_target_length=512 \
  --max_kl_div=0.1 --golden_logits_path=/mnt/disks/persist/golden_qwen35_longprompt.pkl
```
This can be used for running the tests in CI

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: tests/unit/qwen35_partial_mrope_test.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+
+_HEAD_DIM = 256
+_PARTIAL_ROTARY_FACTOR = 0.25
+_ROTARY_DIM = 64
+_MROPE_SECTION = (11, 11, 10)
+_ROPE_THETA = 10_000_000
+
+
+def _reference_partial_mrope(inputs: np.ndarray, positions: np.ndarray) -> np.ndarray:
```
Comment (hengtaoguo) [inline tests/unit/qwen35_partial_mrope_test.py:35]: This test is great. Is there a huggingface equivalent that you could compare against, instead of this numpy implementation? Similar to [qwen3_5_layers_test.py](https://github.com/AI-Hypercomputer/maxtext/blob/main/tests/unit/qwen3_5_layers_test.py). 

Also I wonder if you could move this entire test into the existing qwen3_5_layers_test above.

**Review by hengtaoguo — APPROVED** [review]
Thanks for the fix! Qwen3-Omni and Qwen3.5 are two sets of models (although highly similar in structure). 

Does this suggest Qwen3.5 is reusing `Qwen3OmniMoeThinkerTextRotaryEmbedding` when `use_mrope=True`? Does this bug also exist for Omni model?

File: tests/unit/qwen35_partial_mrope_test.py (hunk 1 above)
Comment (A9isha) [reply tests/unit/qwen35_partial_mrope_test.py:35]: I am working on adding another long prompt test to forward_pass_logit_checker.py with golden logits from HF checked in for CI

**Review by A9isha — COMMENTED** [review]
@hengtaoguo yes `Qwen3OmniMoeThinkerTextRotaryEmbedding` is shared but Omni is not affected because for Omni, `partial_rotary_factor = 1.0` makes `rotary_dim = head_dim`, so "rotate everything" — which is what the unfixed code does — is the correct answer for Omni.

File: tests/unit/qwen35_partial_mrope_test.py (hunk 1 above)
Comment (A9isha) [reply tests/unit/qwen35_partial_mrope_test.py:35]: added forward_pass_logit_checker.py in this PR

**Review by Rohan-Bierneni — APPROVED** [review]
Thank you for this fix! LGTM
